src/app/Shared/Services/Login.service.tsx

```diff
@@ -188,23 +188,64 @@ export class LoginService {
     return this.logout.asObservable();
   }
 
-  setLoggedOut(): void {
-    this.removeCacheItem(this.USER_KEY);
-    this.removeCacheItem(this.TOKEN_KEY);
-    this.token.next('');
-    this.username.next('');
-    this.logout.next();
-    this.sessionState.next(SessionState.NO_USER_SESSION);
-
-    // TODO implement a logout call that will delete the access_token
-    // cached in the oauth server, thus redirecting users to the login page
-    this.queryAuthMethod();
+  setLoggedOut(): Observable<boolean> {
+    return combineLatest([this.getToken(), this.getAuthMethod()]).pipe(
+      first(),
+      concatMap(parts => {
+        const token = parts[0];
+        const method = parts[1];
+
+        return fromFetch(`${this.authority}/api/v2.1/logout`, {
+          credentials: 'include',
+          mode: 'cors',
+          method: 'POST',
+          body: null,
+          headers: this.getAuthHeaders(token, method),
+        });
+      }),
+      concatMap(
+        response => {
+          if(response.status === 302) {
+            const redirectUrl = response.headers.get('X-Location') || '';
```

Comment:
What actually happens here if the headers didn't contain the expected `X-Location`? This defaults to an empty string, and then we try a `fromFetch` on that, which will presumably fail. Would it be better to fail more explicitly by throwing an error if the expected header isn't present?